cognite/seismic/protos/v1/seismic_service_messages.proto

```diff
@@ -169,10 +169,19 @@ message VolumeRequest {
         Identifier seismic = 1;
         int64 seismic_store_id = 2;  // Need to be a data manager or tenant user, not 3rd party, to access by tracestore
     }
-    LineBasedVolume volume = 3;
+    oneof filter {
+        LineBasedVolume volume = 3;
+        GeometryBasedVolume geometry = 5;
+    }
     bool include_trace_header = 4;
 }
 
+message GeometryBasedVolume {
+    Geometry geometry = 1;
+    InterpolationMethod interpolation_method = 2; // Required if the geometry describes a line, otherwise ignored
+    LineDescriptor z_range = 3;
+}
```

Comment:
I like this much better than my approach!

Reply:
I like this interface too. =)